Odata-docs/webapi-8/tutorials/custom-skiptokenhandler.md

```diff
@@ -0,0 +1,549 @@
+
+If you wish to change how skip tokens are generated or how they're processed, you can replace the `DefaultSkipTokenHandler` with your own custom implementation of `SkipTokenHandler`.
+
+## Creating a custom skip token handler
+
+The `SkipTokenHandler` generates the *next link* in an OData response when using server-driven paging. It's also responsible for applying the `$skiptoken` to the results. `SkipTokenHandler` is an abstract class that defines the following methods to be implemented by child classes:
+
+- [`GenerateNextPageLink`](/dotnet/api/microsoft.aspnet.odata.query.skiptokenhandler.generatenextpagelink): generates the `Uri` that's used as the next link in the response
+- [`ApplyTo`](/dotnet/api/microsoft.aspnet.odata.query.skiptokenhandler.applyto): applies the `$skiptoken` to an `IQueryable` collection. This should transform the query to perform the paging. It has two overloads, a generic `ApplyTo<T>` and non-generic `ApplyTo` to handle the generic `IQueryable<T>` and non-generic `IQueryable` respectively.
+
+As an example, let's say we want to represent the skip token using base-64 encoding. This may be useful if we want to "hide" the implementation details and encoding from clients. This is a good use case for a custom `SkipTokenHandler`. We can create our handler from scratch by extending `SkipTokenHandler` directly, or we could extend `DefaultSkipTokenHandler`. The latter is usually more convenient as it allows us to re-use existing logic in `DefaultSkipTokenHandler`. That's the option we'll go with for this sample application.
+
+### Creating and registering the custom handler
+
+Let's create a class called `CustomSkipTokenHandler` that extends `DefaultSkipTokenHandler`. You can put the class anywhere in the project where it makes sense to you.
+
+```csharp
+using Microsoft.AspNetCore.OData.Query;
+
+public class CustomSkipTokenHandler : DefaultSkipTokenHandler
+{
+}
+```
+
+To configure our service to use the `CustomSkipTokenHandler`, we have to inject it in the service container. To do this, we'll make use of an [`AddRouteComponents`](/dotnet/api/microsoft.aspnetcore.odata.odataoptions.addroutecomponents) overload that takes an action that configures services.
+
+Let's rewrite our service configuration to look as follows:
+
+```csharp
+options.SkipToken().AddRouteComponents(
+    "odata",
+    modelBuilder.GetEdmModel(),
+    services => services.AddSingleton<SkipTokenHandler, CustomSkipTokenHandler>())
+```
+
+This tells the service to use `CustomSkipTokenHandler` where a `SkipTokenHandler` is required.
+
+With this configuration, requests that require a `$skiptoken` will go through our `CustomSkipTokenHandler`. But this handler behaves exactly like the default skip token handler because we have not overridden any of its method. Let's start by base64-encoding the skip token when generating the next link.
+
+### Customizing the skip token with `GenerateNextPageLink`
```

Comment:
I thought it may be beneficial to mention the names of the key methods in the heading so that someone skimming through the article can tell which methods are important to override.

Reply:
I ended up adopting your suggestion since I do already list and explain the methods that should be overridden earlier in the document.